src/db.c

```diff
@@ -1871,8 +1871,9 @@ int64_t getAllKeySpecsFlags(struct redisCommand *cmd, int inv) {
  *                               found in other valid keyspecs. 
  */
 int getKeysUsingKeySpecs(struct redisCommand *cmd, robj **argv, int argc, int search_flags, getKeysResult *result) {
-    int j, i, k = 0, last, first, step;
+    int j, i, last, first, step;
     keyReference *keys;
+    result->numkeys = 0;
```

Comment:
IIUC, we should not set it to 0 here, I'm afraid this may cause bugs in the future. Because this function does not hold the `getKeysResult *result` resource. It is the caller's responsibility to initialize or reset the resource, which involves resetting not only the `result->numkeys` but also the `result->keys` and releasing it.

Reply:
I'd love to add an assertion here, in case we forget it, see #12389